tests/functional/adapter/utils/test_utils.py

```diff
@@ -104,3 +105,264 @@ def check_relations_equal(adapter, relation_names: List, compare_snapshot_cols=F
     return check_relations_equal_with_relations(
         adapter, relations, compare_snapshot_cols=compare_snapshot_cols
     )
+
+
+DATALAKE = "rav-test"
+
+
+def base_expected_catalog(
+    project,
+    role,
+    id_type,
+    text_type,
+    time_type,
+    view_type,
+    table_type,
+    model_stats,
+    seed_stats=None,
+    case=None,
+    case_columns=False,
+):
+
+    if case is None:
+
+        def case(x):
+            return x
+
+    col_case = case if case_columns else lambda x: x
```

Comment:
Sorry, what's the purpose of lambda x: x?

Reply:
This was written by dbt, so I'm not entirely sure. Based on my limited interaction with this method, I assume it has to do with the case of specific column types.